### PR TITLE
(CL Test): Add tests for `validateTickInRangeIsValid`

### PR DESCRIPTION
**Pull request by mattverse**
Closes: https://github.com/osmosis-labs/osmosis/issues/3983

## What is the purpose of the change
Adds unit tests for `validateTickInRangeIsValid` method

## Testing and Verifying
This PR adds new tests.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/tick_test.go [hunk 1]
```diff
@@ -551,3 +551,86 @@ func (s *KeeperTestSuite) TestGetLiquidityDepthFromIterator() {
...
+			upperTick: 4,
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			s.Setup()
+
+			// use default exponent at price one
+			exponentAtPriceOne := DefaultExponentAtPriceOne
```
Comment (p0mvn) [inline x/concentrated-liquidity/tick_test.go:621]: Consider adding one test case where this exponent is different to make sure that the success of the function is independent of this value

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/tick_test.go (hunk 1 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/tick_test.go:621]: Agreed, would just make `exponentAtPriceOne` a test param

File: x/concentrated-liquidity/tick_test.go [hunk 2]
```diff
@@ -551,3 +551,86 @@ func (s *KeeperTestSuite) TestGetLiquidityDepthFromIterator() {
...
+
+			tickSpacing := uint64(2)
+
+			// System Under Test
+			err := cl.ValidateTickInRangeIsValid(tickSpacing, exponentAtPriceOne, test.lowerTick, test.upperTick)
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/tick_test.go:626]: Why not use defaultTickSpacing you set at the top of this test?

**Review by czarcas7ic — APPROVED** [review]
LGTM

File: x/concentrated-liquidity/tick_test.go (hunk 2 above)
Comment (mattverse) [reply x/concentrated-liquidity/tick_test.go:626]: switched!

File: x/concentrated-liquidity/tick_test.go (hunk 1 above)
Comment (mattverse) [reply x/concentrated-liquidity/tick_test.go:642]: https://github.com/osmosis-labs/osmosis/pull/4240/commits/719c993d229e1caae4a23514185721e9033a22c6

**Pull request merged**